cli-plugins/socket/socket_noabstract.go

```diff
@@ -3,23 +3,12 @@
 package socket
 
 import (
-	"net"
 	"os"
 	"path/filepath"
-	"syscall"
 )
 
-func listen(socketname string) (*net.UnixListener, error) {
-	// Because abstract sockets are unavailable, we create a socket in the
-	// system temporary directory instead.
-	return net.ListenUnix("unix", &net.UnixAddr{
-		Name: filepath.Join(os.TempDir(), socketname),
-		Net:  "unix",
-	})
-}
-
-func unlink(listener *net.UnixListener) {
-	// unlink(2) is best effort here; if it fails, we may 'leak' a socket
-	// into the filesystem, but this is unlikely and overall harmless.
-	_ = syscall.Unlink(listener.Addr().String())
+func socketName(basename string) string {
+	// Because abstract sockets are unavailable, use a socket path in the
+	// system temporary directory.
+	return filepath.Join(os.TempDir(), basename)
```

Comment:
Can we add a comment explaining why this one doesn't start with `@` while the other one does?

Reply:
I thought it was self-evident from the buildtags -- this is the version for platforms that don't support abstract sockets. The inverse is also true; the abstract socket version uses the `@` to signify an abstract socket on supported platforms...

Reply:
It's not self-evident that the socket will be automatically unlinked just because it doesn't start with a `@` though.

Reply:
That's not relevant in this file -- that's a convention of the Go `net` API. I've linked to the documentation in the commit message, but this shouldn't require being explicitly called out. The fact that Go unlinks the socket file on `Close()` is a well-known part of the API and a 'normal' usage.

Reply:
I've tried added a comment to `socket.go`, since the main issue here seems to be a desire for documentation of the implicit unlink that is part of the net package.